aspnetcore/hosting/ihostingstartup.md

```diff
@@ -0,0 +1,164 @@
+
+[!code-csharp[Main](ihostingstartup/sample/HostingStartupSample/Pages/Index.cshtml.cs?name=snippet1&highlight=14-16)]
+
+## Disable automatic loading of hosting startup assemblies
+
+There are two ways to disable the automatic loading of hosting startup assemblies:
+
+* Set the [Prevent Hosting Startup](xref:fundamentals/hosting#prevent-hosting-startup) host configuration setting.
+* Set the `ASPNETCORE_preventHostingStartup` environment variable.
+
+When either the host setting or the environment variable is set to `true` or `1`, hosting startup assemblies aren't automatically loaded. If both are set, the host setting controls the behavior.
+
+Disabling hosting startup assemblies using the host setting or environment variable disables them globally and may disable several features of an app. It isn't currently possible to selectively disable a hosting startup assembly added by a library unless the library offers its own configuration option. A future release will offer the ability to selectively disable hosting startup assemblies (see [GitHub issue aspnet/Hosting #1243](https://github.com/aspnet/Hosting/pull/1243)).
+
+## Implement IHostingStartup features
+
+### Create the assembly
+
+An `IHostingStartup` feature is deployed as an assembly based on a console app without an entry point. The assembly references the [Microsoft.AspNetCore.Hosting.Abstractions](https://www.nuget.org/packages/Microsoft.AspNetCore.Hosting.Abstractions/) package:
+
+[!code-xml[Main](ihostingstartup/snapshot_sample/StartupFeature.csproj)]
+
+A [HostingStartup](/dotnet/api/microsoft.aspnetcore.hosting.hostingstartupattribute) attribute identifies a class as an implementation of `IHostingStartup` for loading and execution when building the [IWebHost](/dotnet/api/microsoft.aspnetcore.hosting.iwebhost). In the following example, the namespace is `StartupFeature`, and the class is `StartupFeatureHostingStartup`:
+
+[!code-csharp[Main](ihostingstartup/snapshot_sample/StartupFeature.cs?name=snippet1)]
+
+A class implements `IHostingStartup`. The class's [Configure](/dotnet/api/microsoft.aspnetcore.hosting.ihostingstartup.configure) method uses an [IWebHostBuilder](/dotnet/api/microsoft.aspnetcore.hosting.iwebhostbuilder) to add features to an app:
+
+[!code-csharp[Main](ihostingstartup/snapshot_sample/StartupFeature.cs?name=snippet2&highlight=3,5)]
+
+When building an `IHostingStartup` project, the dependencies file (*\*.deps.json*) sets the `runtime` location of the assembly to the *bin* folder:
+
+[!code-json[Main](ihostingstartup/snapshot_sample/StartupFeature1.deps.json?range=2-13&highlight=8)]
+
+Only part of the file is shown. The assembly name in the example is `StartupFeature`.
+
+### Update the dependencies file
+
+In order for the feature to activate for an app, the `runtime` location in the *\*.deps.json* file must be changed to reflect the location of the feature's runtime assembly in the runtime store. Prefix the `runtime` location with `lib/netcoreapp2.0/`:
```

Comment:
Way too long. Alt:
The `runtime` location for features is specified in the *\*.deps.json* file. To active a feature,  the `runtime` must include the location of the feature's runtime assembly. Prefix the `runtime` location with `lib/netcoreapp2.0/`:

Or can you make a bullet list?

Reply:
I'll try a shorter version of this that breaks the sentence as you suggest.